pkg/cluster/self_hosted.go

```diff
@@ -46,6 +46,71 @@ func (c *Cluster) selectMasterNodes() ([]string, error) {
 	return res, nil
 }
 
+func (c *Cluster) waitNewMember(oldN, retries int, name string) error {
+	return retryutil.Retry(10*time.Second, retries, func() (bool, error) {
+		err := c.updateMembers(c.members)
+		if err != nil {
+			c.logger.Warningf("unable to update members: %v", err)
+			return false, nil
+		}
+		if c.members.Size() > oldN {
+			return true, nil
+		}
+		c.logger.Infof("still waiting for the new self hosted member (%s) to start...", name)
+		return false, nil
+	})
+}
+
+func (c *Cluster) changePodToNoopIfNotScheduled(name, ns string) string {
```

Comment:
We should probably return both.
But since empty nodename could be used so returning nodename is good enough.